pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -173,55 +178,33 @@ type libInfo struct {
 // extractLibInfo returns the language, the image,
 // and a boolean indicating whether the library should be injected into the pod
 func extractLibInfo(pod *corev1.Pod, containerRegistry string) []libInfo {
-	libInfoList := []libInfo{}
-	podAnnotations := pod.GetAnnotations()
-	for _, lang := range supportedLanguages {
-		customLibAnnotation := strings.ToLower(fmt.Sprintf(customLibAnnotationKeyFormat, lang))
-		if image, found := podAnnotations[customLibAnnotation]; found {
-			libInfoList = append(libInfoList, libInfo{
-				lang:  lang,
-				image: image,
-			})
-		}
+	libInfoMap := map[string]libInfo{}
 
-		libVersionAnnotation := strings.ToLower(fmt.Sprintf(libVersionAnnotationKeyFormat, lang))
-		if version, found := podAnnotations[libVersionAnnotation]; found {
-			image := fmt.Sprintf("%s/dd-lib-%s-init:%s", containerRegistry, lang, version)
-			libInfoList = append(libInfoList, libInfo{
-				lang:  lang,
-				image: image,
-			})
+	// Inject all libraries if Single Step Instrumentation is enabled
+	if isApmInstrumentationEnabled(pod.Namespace) {
+		libInfoMap = injectAll(pod.Namespace, containerRegistry)
+		if len(libInfoMap) > 0 {
+			log.Debugf("Single Step Instrumentation: Injecting all libraries into pod %q in namespace %q", podString(pod), pod.Namespace)
 		}
+	}
 
-		for _, ctr := range pod.Spec.Containers {
-			customLibAnnotation := strings.ToLower(fmt.Sprintf(customLibAnnotationKeyCtrFormat, ctr.Name, lang))
-			if image, found := podAnnotations[customLibAnnotation]; found {
-				libInfoList = append(libInfoList, libInfo{
-					ctrName: ctr.Name,
-					lang:    lang,
-					image:   image,
-				})
-			}
-
-			libVersionAnnotation := strings.ToLower(fmt.Sprintf(libVersionAnnotationKeyCtrFormat, ctr.Name, lang))
-			if version, found := podAnnotations[libVersionAnnotation]; found {
-				image := fmt.Sprintf(imageFormat, containerRegistry, lang, version)
-				libInfoList = append(libInfoList, libInfo{
-					ctrName: ctr.Name,
-					lang:    lang,
-					image:   image,
-				})
-			}
-		}
+	// If admission.datadoghq.com/enabled: "true" or mutateUnlabelled is set,
+	// the library version specified via annotation on the Pod takes precedence
```

Comment:
🤔 this is interesting since it means that if someone puts library label on the deployment (and not the enabled annotation) and uses Single Step then the library label will be ignored.

I think it will reduce confusion to respect the library label for Single Step regardless of the enabled annotation and mutateUnlabelled setting.

Reply:
Yea I agree it makes more sense. Going to update the code.